osquery/sql/tests/sql.cpp

```diff
@@ -314,14 +314,12 @@ TEST_F(SQLTests, test_regex_split_slashes) {
   query(
       "select regex_split('/foo/bar', '/', 0) as t0, \
                 regex_split('/foo/bar', '/', 1) as t1, \
-                regex_split('/foo/bar', '/', 2) as t2, \
-                regex_split('/foo/bar', '/', 3) as t3",
+                regex_split('/foo/bar', '/', 2) as t2",
       d);
   ASSERT_EQ(d.size(), 1U);
-  EXPECT_EQ(d[0]["t0"], "");
-  EXPECT_EQ(d[0]["t1"], "foo");
-  EXPECT_EQ(d[0]["t2"], "bar");
-  EXPECT_EQ(d[0]["t3"], "");
+  EXPECT_EQ(d[0]["t0"], "foo");
```

Comment:
This feels like a breaking change. And I have no idea what correct is in the larger world.

Though, I guess `regex_split` and `split` behave differently, and that's also confusing:
```
osquery> select split('/foo/bar', '/', 1), regex_split('/foo/bar', '/', 1);
+---------------------------+---------------------------------+
| split('/foo/bar', '/', 1) | regex_split('/foo/bar', '/', 1) |
+---------------------------+---------------------------------+
| bar                       | foo                             |
+---------------------------+---------------------------------+
```

Still, I worry about a breaking change